### ISSUE TITLE
clarify loss functions for TextCategorizer and Parser?

### ISSUE THREAD

**Issue by switchstard** [explosion/spaCy#6383]
Great library! I'm trying to understand it better, in particular the loss functions it uses, and came across the following issues which explained that for most components including the TextCategorizer the loss is log-loss:
https://github.com/explosion/spaCy/issues/5392
https://github.com/explosion/spaCy/issues/5537
https://github.com/explosion/spaCy/issues/1869

However I'm confused because I'm not finding the log loss function calculation in the code. When I look at the code in text categorizer here:
https://github.com/explosion/spaCy/blob/320a8b14814c7e0c6dce705ad7bf0f13bf64b61c/spacy/pipeline/pipes.pyx#L1041-L1055
it is computing the mean square error as the loss. 

Also I'm not sure I understand the cpu_log_loss function here
https://github.com/explosion/spaCy/blob/320a8b14814c7e0c6dce705ad7bf0f13bf64b61c/spacy/syntax/_parser_model.pyx#L180-L203
It looks similar to a soft-max computation.

Would appreciate if someone could help point me to where the log loss functions are calculated or if I'm missing something here. Thanks!

**Comment by svlandeg** [comment]
The code snippets you linked, are indeed where the losses are being calculate - generally these are `get_loss` functions of objects implementing the `TrainablePipe` API.

The reference to a "log loss" was only mentioned in one of the three issues you linked, which is from 2018, so I'd imagine that this changed since then.

**Comment by switchstard** [comment]
> The reference to a "log loss" was only mentioned in one of the three issues you linked, which is from Jan 2018, so I'd imagine that this changed since then.

It's also mentioned in https://github.com/explosion/spaCy/issues/5537 from June 2020 (cross-entropy loss = log loss). Is that mention incorrect?

For the cpu_log_loss function is there a name for the loss that it's computing?

**Comment by svlandeg** [comment]
Actually, rereading those posts, I wonder whether Matt didn't mean to say that it's the parser that uses the "multi-label log loss", as he calls it. I've been trying to track the `textcat` code but I couldn't find any evidence of the log function changing. 

And the "multi-label log loss" is indeed similar to a softmax, as you mentioned.

Either way - the code implementations really are the only source of truth for this... :|
